examples/nlp/language_modeling/megatron_quantization.py

```diff
@@ -64,7 +70,18 @@ def main(cfg) -> None:
     if not torch.cuda.is_available():
         raise EnvironmentError("GPU is required for the inference.")
 
-    quantizer = Quantizer(cfg.quantization, cfg.inference, cfg.export, cfg.trainer)
+    # Overwrite model config with the one from the model checkpoint and apply quantization modifications
+    model_cfg = load_config(cfg.model.restore_from_path)
+    with open_dict(model_cfg):
+        for key, val in cfg.model.items():
+            model_cfg[key] = val
+    model_cfg = Quantizer.modify_model_config(model_cfg)
```

Comment:
Since L75-77 also modify model config, they could become a part of `modify_model_config` to keep it all in a single location

Reply:
SFTModel already has a utility function doing this called [`merge_cfg_with`](https://github.com/NVIDIA/NeMo/blob/main/nemo/collections/nlp/parts/mixins/nlp_adapter_mixins.py#L470) hence I kept it separate

Reply:
1. Could we move overriding fields with `cfg.model` to ` Quantizer.modify_model_config` anyway? We modify config, let's do it in one place to keep things organized i.e. that `modify_model_config` method
2. [minor] Can we use https://omegaconf.readthedocs.io/en/2.3_branch/usage.html#omegaconf-update instead of manual loop in L75-77